src/systemctl/systemctl.c

```diff
@@ -7336,6 +7341,53 @@ static int unit_file_create_new(
         if (r < 0)
                 return r;
 
+        if (unit_paths) {
+                _cleanup_free_ char *tmp_contents = NULL;
+                _cleanup_fclose_ FILE *f = NULL;
+                char **path;
+                size_t size;
+
+                r = read_full_file(tmp_path, &tmp_contents, &size);
+                if (r < 0)
+                        return log_error_errno(errno, "Failed to open \"%s\": %m", tmp_path);
```

Comment:
read_full_file() returns the error in `r`, not `errno`, hence please pass `r` as first arg to log_error_errno()